### PR TITLE
Update accessibility.mdx

### PR DESCRIPTION
**Pull request by mbgower**
Updating the information on the accessibility tab of list to the new template and content.

Addresses content evaluated in https://ibm.box.com/s/94q54gsl3tac5hf51hr4h93n7aogryxn

### CODE REVIEW

File: src/pages/components/list/accessibility.mdx [hunk 1]
```diff
@@ -17,353 +17,26 @@ import {
...
-          </StructuredListCell>
-        </StructuredListRow>
-      </StructuredListBody>
-    </StructuredListWrapper>
-  </Column>
-</Row>
+- Carbon uses native [HTML](https://www.w3.org/WAI/tutorials/page-structure/content/#lists) ordered (`ol`) and unorder (`ul`) lists and list items
+(`li`), then styles them with CSS.
+- Carbon uses `::before` and `::marker` CSS pseudo-elements for the numbering
+and bulleting of lists, which are properly read by assistive technologies such as [screen readers](https://www.ibm.com/able/toolkit/verify/screen-reader/).
```
Comment (aagonzales) [inline src/pages/components/list/accessibility.mdx:42]: There was a weird line break here

```suggestion
- Carbon uses `::before` and `::marker` CSS pseudo-elements for the numbering and bulleting of lists, which are properly read by assistive technologies such as [screen readers](https://www.ibm.com/able/toolkit/verify/screen-reader/).
```

File: src/pages/components/list/accessibility.mdx [hunk 2]
```diff
@@ -17,353 +17,26 @@ import {
...
 
-This component has been validated to meet the
-[WCAG 2.0 AA](https://www.w3.org/TR/WCAG20/) and
-[Section 508](http://www.section508.gov/) accessibility guidelines, however
-changes made by the content owner can affect accessibility compliance. Be sure
-to review and follow the guidance in this section when updating or adding new
-content to this component.
+Lists are not keyboard operable, unless the list items themselves are operable. In
+such a situation, the list items will retain the component’s default keyboard
+interaction. For example, in a list of links, each [link](https://carbondesignsystem.com/components/link/usage) will be in the tab order and can be activated by Enter.
```
Comment (aagonzales) [inline src/pages/components/list/accessibility.mdx:33]: "Enter" needs to be in inline code snippet format
```suggestion
interaction. For example, in a list of links, each [link](https://carbondesignsystem.com/components/link/usage) will be in the tab order and can be activated by `Enter`.
```

**Review by aagonzales — COMMENTED** [review]
Looks good, just two formatting bugs

**Review by dakahn — APPROVED** [review]

**Comment by dakahn** [comment]
pushed a prettier fix @mbgower -- CI should be good now 👍🏾

File: src/pages/components/list/accessibility.mdx [hunk 3]
```diff
@@ -17,353 +17,31 @@ import {
...
-        </StructuredListRow>
-      </StructuredListBody>
-    </StructuredListWrapper>
-  </Column>
-</Row>
+- Carbon uses native
+  [HTML](https://www.w3.org/WAI/tutorials/page-structure/content/#lists) ordered
+  (`ol`) and unorder (`ul`) lists and list items (`li`), then styles them with
+  CSS.
+- Carbon uses `::before` and `::marker` CSS pseudo-elements for the numbering
```
[the remark this answers is not in the document]
Comment (tay1orjones) [reply src/pages/components/list/accessibility.mdx]: I think we have a bug with the list item implementation. I'll work on a fix

**Review by tay1orjones — APPROVED** [review]
@mbgower @aagonzales I have a fix for the link spacing issue coming in https://github.com/carbon-design-system/carbon-website/pull/2996.

We can merge this as-is if you'd like, that PR will fix it once it's in.

**Review by aagonzales — APPROVED** [review]

**Review by abbeyhrt — APPROVED** [review]

**Pull request merged**